pandora-common/src/gameLogic/locks/lockData.ts

```diff
@@ -9,10 +9,17 @@ export const LockDataBundleSchema = z.object({
 		name: z.string(),
 		/** Time the item was locked */
 		time: z.number(),
-		/** Time the timer on the lock will expire, if lock includes a timer */
-		lockedUntil: z.number().int().nonnegative().optional(),
 	}).optional(),
+	/** Data applicable only to locks with timer */
+	timer: z.object({
+		/** Time the timer on the lock will expire */
+		lockedUntil: z.number().int().nonnegative(),
+		/** Whether the character that locked it is allowed to unlock the timer early */
+		allowEarlyUnlock: z.boolean(),
+	}).optional(),
```

Comment:
This change breaks backwards compatibility without writing some additional migration code.
You should either write a migration code for this (which could be done using `z.preprocess` - look at `SpacePublicSettingSchema` for example).
Alternatively it could be done in a backward-compatible way in the first place by preserving the `lockedUntil` property as it is and introducing a new property along the lines of `disallowEarlyUnlock: z.literal(true).optional()` - where it missing is the current behavior.
I also wonder if the timer data shouldn't only be present in the `locked` section. Does it make sense for unlocked locks at all?